tests/HttpClientTest.cpp

```diff
@@ -108,105 +100,97 @@ static int s_TestHttpDownloadNoBackPressure(struct aws_allocator *allocator, voi
-        std::lock_guard<std::mutex> lockGuard(semaphoreLock);
-
-        connectionShutdown = true;
-        if (errorCode)
-        {
-            errorOccured = true;
-        }
-
-        semaphore.notify_one();
-    };
-
-    Http::HttpClientConnectionOptions httpClientConnectionOptions;
-    httpClientConnectionOptions.allocator = allocator;
-    httpClientConnectionOptions.bootstrap = &clientBootstrap;
-    httpClientConnectionOptions.onConnectionSetup = onConnectionSetup;
-    httpClientConnectionOptions.onConnectionShutdown = onConnectionShutdown;
-    httpClientConnectionOptions.socketOptions = &socketOptions;
-    httpClientConnectionOptions.tlsConnOptions = &tlsConnectionOptions;
-    httpClientConnectionOptions.initialWindowSize = SIZE_MAX;
-    httpClientConnectionOptions.hostName = hostName;
-    httpClientConnectionOptions.port = 443;
-
-    std::unique_lock<std::mutex> semaphoreULock(semaphoreLock);
-    ASSERT_TRUE(Http::HttpClientConnection::CreateConnection(httpClientConnectionOptions));
-    semaphore.wait(semaphoreULock, [&]() { return connection || connectionShutdown; });
-
-    ASSERT_FALSE(errorOccured);
-    ASSERT_FALSE(connectionShutdown);
-    ASSERT_TRUE(connection);
-
-    int responseCode = 0;
-    std::ofstream downloadedFile("http_download_test_file.txt", std::ios_base::binary);
-    ASSERT_TRUE(downloadedFile);
-
-    Http::HttpRequestOptions requestOptions;
-    requestOptions.onStreamOutgoingBody = nullptr;
-
-    bool streamCompleted = false;
-    requestOptions.onStreamComplete = [&](Http::HttpStream &stream, int errorCode) {
-        std::lock_guard<std::mutex> lockGuard(semaphoreLock);
-
-        streamCompleted = true;
-        if (errorCode)
-        {
-            errorOccured = true;
-        }
-
-        semaphore.notify_one();
-    };
-    requestOptions.onIncomingHeadersBlockDone = nullptr;
-    requestOptions.onIncomingHeaders = [&](Http::HttpStream &stream, const Http::HttpHeader *header, std::size_t len) {
-        responseCode = stream.GetResponseStatusCode();
-    };
-    requestOptions.onIncomingBody = [&](Http::HttpStream &, const ByteCursor &data, std::size_t &outWindowUpdateSize) {
-        downloadedFile.write((const char *)data.ptr, data.len);
-    };
-
-    requestOptions.method = ByteCursorFromCString("GET");
-    requestOptions.uri = uri.GetPathAndQuery();
-
-    Http::HttpHeader host_header;
-    host_header.name_str = ByteCursorFromCString("host");
-    host_header.value = uri.GetHostName();
-    requestOptions.headerArray = &host_header;
-    requestOptions.headerArrayLength = 1;
-
-    connection->NewClientStream(requestOptions);
-    semaphore.wait(semaphoreULock, [&]() { return streamCompleted; });
-    ASSERT_INT_EQUALS(200, responseCode);
-
-    connection->Close();
-    semaphore.wait(semaphoreULock, [&]() { return connectionShutdown; });
-
-    downloadedFile.flush();
-    downloadedFile.close();
+    {
```

Comment:
don't modify HttpClientTest.cpp to use the ConnectionManager

Reply:
Will revert